File: python/special_order.py
import math

def count_special_order(n: int):
    results = []
    m = math.ceil(math.sqrt(n))
    for i in range(m * m):
        digit1, digit2 = divmod(i, m)
        num_str = f"{digit1}{digit2}"
        results.append((num_str, digit1 + digit2, digit1, digit2))

    results.sort(key=lambda x: (x[1], x[2], x[3]))
    return [r[0] for r in results]

File: csharp/SpecialOrder.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class SpecialOrder
{
    public static List<string> CountSpecialOrder(int n)
    {
        var results = new List<(string num, int sum, int d1, int d2)>();
        int m = (int)Math.Ceiling(Math.Sqrt(n));

        for (int i = 0; i < m * m; i++)
        {
            int d1 = i / m;
            int d2 = i % m;
            string numStr = $"{d1}{d2}";
            results.Add((numStr, d1 + d2, d1, d2));
        }

        return results
            .OrderBy(r => r.sum)
            .ThenBy(r => r.d1)
            .ThenBy(r => r.d2)
            .Select(r => r.num)
            .ToList();
    }
}
